pkg/workflow/workflow.go

```diff
@@ -270,79 +290,88 @@ func (w *workflow) Trace() error {
 	return recorder.With(w.cli, w.app).Save("", data).Limit(10).Error()
 }
 
-func (w *workflow) GetBackoffWaitTime() time.Duration {
-	nextTime, ok := w.wfCtx.GetValueInMemory(wfTypes.ContextKeyNextExecuteTime)
-	if !ok {
-		return time.Second
+func (w *workflow) GetSuspendBackoffWaitTime() time.Duration {
+	if w.app.Status.Workflow == nil {
+		return 0
 	}
-	unix, ok := nextTime.(int64)
-	if !ok {
-		return time.Second
+	stepStatus := make(map[string]common.StepStatus)
+	for _, ss := range w.app.Status.Workflow.Steps {
+		setStepStatus(stepStatus, ss.StepStatus)
+		for _, sss := range ss.SubStepsStatus {
+			setStepStatus(stepStatus, sss.StepStatus)
+		}
 	}
-	next := time.Unix(unix, 0)
-	if next.After(time.Now()) {
-		return time.Until(next)
+	max := time.Duration(1<<63 - 1)
+	min := max
+	if w.app.Spec.Workflow != nil {
+		for _, step := range w.app.Spec.Workflow.Steps {
+			if step.Type == wfTypes.WorkflowStepTypeSuspend || step.Type == wfTypes.WorkflowStepTypeStepGroup {
+				min = handleSuspendBackoffTime(step, stepStatus[step.Name], min)
+			}
+			for _, sub := range step.SubSteps {
+				if sub.Type == wfTypes.WorkflowStepTypeSuspend {
+					min = handleSuspendBackoffTime(oamcore.WorkflowStep{
+						Name:       sub.Name,
+						Type:       sub.Type,
+						Timeout:    sub.Timeout,
+						Properties: sub.Properties,
+					}, stepStatus[sub.Name], min)
+				}
+			}
+		}
 	}
-
-	return time.Second
+	if min == max {
+		return 0
+	}
+	return min
 }
 
-func (w *workflow) HandleSuspendWait(ctx monitorContext.Context) (doWaiting bool, durationWaiting time.Duration, errRet error) {
-	ctx.Info("handle suspend wait")
```

Comment:
shall we keep this kind of info？I don't know if it's helpful, we should clean them if they're useless.

Reply:
cc @leejanee